File: BISYNC.java
import java.io.*;
import java.nio.file.*;

public class BISYNC {
  public static void printByteCodes(byte[] data) {
    int n = 0;
    for (byte b : data) {
      String st = String.format("%02X ", b);
      System.out.print(st);
      if(n % 16 == 7)
        System.out.print(" | ");
      if(n % 16 == 15)
        System.out.println();
      n++;
    }
  }
  
  public static byte[] bisync_body(byte[] data){
    byte[] body = {};
    // put your code here

    return body;
  }

  public static void main(String[] args){
    if(args.length < 2)
      System.out.println("Usage: java BISYNC [input path] [output path]");
    else{
      String inputPath = args[0];
      String outputPath = args[1];

      byte[] array = {};
      byte[] body = {};
      // Read data
      try{
        array = Files.readAllBytes(Paths.get(inputPath));
      } catch (IOException e) {
        e.printStackTrace();
      }
      System.out.println("\nData");
      printByteCodes(array);

      // Convert data to BISYNC body
      body = bisync_body(array);
      
      System.out.println("\nBody");
      printByteCodes(body);

      // Write body
      try{
        Files.write(Paths.get(outputPath), body, StandardOpenOption.CREATE);
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
  }
}

File: BISYNC.py
import sys

def print_byte_codes(data):
  for n, c in enumerate(data):
    print(f"{c:02X}", end=" ")
    if n % 16 == 7:
      print(" | ", end="")
    if n % 16 == 15:
      print()

def bisync_body(data):
  body = b""
  #put your code here
  return body

if __name__ == "__main__":
  # execute only if run as a script
  if len(sys.argv) < 3:
    print(f"Usage: {sys.argv[0]} [input path] [output path]")
  else:
    input_path = sys.argv[1]
    output_path = sys.argv[2]
    
    # read data
    with open(input_path, "rb") as file:
      byte = file.read()
    print("\nData")
    print_byte_codes(byte)

    # convert data to BISYNC body
    body = bisync_body(byte)

    print("\nBody")
    print_byte_codes(body)
    # write body
    with open(output_path, "wb") as file:
      file.write(body)
